src/parser/parser_utils.js

```diff
@@ -187,6 +188,39 @@ function splitVAST(ads) {
   return splittedVAST;
 }
 
+/**
+ * Parses the attributes and assign them to object
+ * @param  {Object} attributes attribute
+ * @param  {Object} verificationObject with properties which can be assigned
+ * @return {AdVerification|null}
+ */
+function assignAttributes(attributes, verificationObject) {
```

Comment:
at first glance it looks like this function can be re-used, but actually it only applies to `AdVerification` class. you could either specify that this function is only for `AdVerification` or pass the class as an argument

Reply:
Sorry, I'm not sure what do you mean exactly.

Reply:
If I understand correctly, this function checks that the attribute exists in the class `AdVerification`, and if so then assign the value to it. you could do:
```
function assignAttributes(attributes, verificationObject, verificationClassDefault) {
  // ...
  const ver = Object.assign(new verificationClassDefault, verificationObject)
}

assignAttributes(node.attributes, verification, new AdVerification)
```

But actually I don't quite understand the purpose of line 202, why is there a need to assign `AdVerification` to `verificationObject`?

Reply:
Well it was your idea to clone it, no?
https://github.com/dailymotion/vast-client-js/pull/297#discussion_r305354981

Reply:
Yes, my point is `verificationObject` is already an instance of `AdVerification`, you don't have to assign it again to `AdVerification`. A simple: `const ver = { ...verificationObject }` would do.

Currently you have to assign the returned class instance of this function to where you call it, so perhaps it would make the most sense to re-use the function `parseAttributes`, and directly pass them on creating `new AdVerification()` like in https://github.com/dailymotion/vast-client-js/blob/3.0-version/src/parser/creative_companion_parser.js#L23 . This way you are only returning the attributes and not the class instance.
What do you think?

Reply:
From the second thought, I decided to revert last change and actually mutate the object.
The reason is that I use this object twice
https://github.com/dailymotion/vast-client-js/blob/e0a75738601e9253dda97d6a1bb96c0e5297c3a2/src/parser/ad_parser.js#L329
and https://github.com/dailymotion/vast-client-js/blob/e0a75738601e9253dda97d6a1bb96c0e5297c3a2/src/parser/ad_parser.js#L339

So it easier to just mutate. It gives me a simple type check as well if I use real class.

I disagree, then let's start a discussion from this point again )

Reply:
this is ok too as it is simpler

Reply:
If disagree*